autobump-maven/src/test/java/com/github/autobump/maven/model/MavenDependencyResolverTest.java

```diff
@@ -29,18 +29,51 @@ public void TestSuccesresolve() {
         Set<Dependency> deps = dependencyResolver.resolve(workspace);
         assertEquals(
                 Set.of(Dependency.builder()
-                .group("org.apache.derby")
-                .name("derby")
-                .version("10.15.2.0")
-                .build()),
+                        .group("org.apache.derby")
+                        .name("derby")
+                        .version("10.15.2.0")
+                        .build()),
+                deps);
+    }
+
+    @Test
+    public void TestSuccesresolveProperties() {
+        Workspace ws = new Workspace("src/test/resources/project_root_support_properties");
+        Set<Dependency> deps = dependencyResolver.resolve(ws);
+        assertEquals(
+                Set.of(Dependency.builder()
+                        .group("org.apache.derby")
+                        .name("derby")
+                        .version("10.15.2.0")
+                        .build()),
+                deps);
+    }
+
+    @Test
+    public void TestresolveNullProperty() {
+        assertThrows(DependencyParserException.class, () ->
```

Comment:
I would argue that if a dependency is defined using a version property: ${...} and we cannot find the property, that it is best for us to ignore the dependency instead of throwing an exception and stopping the entire dependency resolution.

There could be cases where the developer uses ${project.version} for example to define the version of the dependency. The project.version will not be found in the <properties/> section but it is still perfectly valid.

In Maven, it is also possible to provide property values from the command line. E.g. devs could run: `mvn package -Dorg.apache.derby.version=1.0`. Again, the property is not defined in the `pom.xml` file but it is still valid.

So bottom line, I think it's safer to ignore unresolved properties instead of throwing exceptions.